tests/test_exposition.py

```diff
@@ -330,6 +354,16 @@ def my_auth_handler(url, method, timeout, headers, data):
         self.assertEqual(self.requests[0][0].headers.get('content-type'), CONTENT_TYPE_LATEST)
         self.assertEqual(self.requests[0][1], b'# HELP g help\n# TYPE g gauge\ng 0.0\n')
 
+    def test_push_with_redirect_handler(self):
+        def my_redirect_handler(url, method, timeout, headers, data):
+            return passthrough_redirect_handler(url, method, timeout, headers, data)
+
+        push_to_gateway(self.address, "my_job_with_redirect", self.registry, handler=my_redirect_handler)
+        self.assertEqual(self.requests[0][0].command, 'PUT')
+        self.assertEqual(self.requests[0][0].path, '/metrics/job/my_job_with_redirect')
+        self.assertEqual(self.requests[0][0].headers.get('content-type'), CONTENT_TYPE_LATEST)
+        self.assertEqual(self.requests[0][1], b'# HELP g help\n# TYPE g gauge\ng 0.0\n')
```

Comment:
Should there also be assertions for the second(redirected) request?

Reply:
fixed in https://github.com/prometheus/client_python/pull/622/commits/912cb183ed6263e272869fb0b571e36c55e3073e

Let me know if you feel this is enough coverage, or if there's a different approach to take which can cover the test better.

Reply:
Adding the extra test revealed some further complexity about how to handle `Request` objects in Python 2.7. It looks quite hairy, no simple solution. I will pick it back up tomorrow.

Reply:
update: I was just making a dumb mistake on the Python 2.7 version check. I fixed that and it seems to have resolved the error, so I think it is ready for a new round of review.

Reply:
I think this test looks reasonable.